hw/ip/otbn/dv/rig/rig/gens/bad_branch.py

```diff
@@ -99,24 +97,26 @@ def gen(self,
         assert op_val_grs1 is not None
 
         # We need to pick an out of bounds offset from all available values (A)
-        # Best way to solve this is exclude the set of valid choices (B)
-        # We know tgt_addr can be max : pc + max_offset (A_max)
-        # We know tgt_addr can be min : pc + min_offset (A_min)
-        # Aim: (PC + min_offset, PC + max_offset) - (0, imem_size)
-
-        # Choose target from (A_min, A_max-B_max)
-        local_max = max_offset - program.imem_size
-        tgt_addr = random.randrange(min_offset, local_max, 2)
-
-        # If chosen value is in B, push it out by adding B_max
-        if tgt_addr >= 0:
-            tgt_addr += program.imem_size
-        assert tgt_addr < 0 or tgt_addr > program.imem_size
-
+        # mode is a random variable to ensure that all the bins are hit
+        # for offset range. It gives equal weight to all address ranges.
+
+        mode = random.randint(0, 4)
+        if mode == 0 and min_addr <= -4:
+            tgt_addr = random.randrange(min_addr, -4, 4)
+        elif mode <= 1 and program.imem_size <= max_addr:
+            tgt_addr = random.randrange(program.imem_size, max_addr, 4)
+        elif mode <= 2 and min_addr <= -4:
+            tgt_addr = min_addr
+        elif mode <= 3 and program.imem_size <= max_addr:
+            tgt_addr = max_addr
+        else:
+            assert mode <= 3
+            tgt_addr = random.randrange(min_addr + 2, max_addr, 4)
```

Comment:
Can you maybe add another mode for in range but misaligned (meaning not divisible by 4)  `tgt_addr` so that we would also cover misaligned branches (#8086)?